extensions/binding/org.eclipse.smarthome.binding.fsinternetradio/src/main/java/org/eclipse/smarthome/binding/fsinternetradio/internal/FSInternetRadioDiscoveryParticipant.java

```diff
@@ -100,28 +100,13 @@ public DiscoveryResult createResult(RemoteDevice device) {
                 }
 
                 final DiscoveryResult result = DiscoveryResultBuilder.create(uid).withProperties(properties)
-                        .withLabel(getLabel(device)).build();
+                        .withLabel(device.getDisplayString()).build();
```

Comment:
Well, it's been a long code journey here :) I hope I can explain it.
1. `createResult(RemoteDevice device)`  returns something different from null only if `getThingUID(RemoteDevice device) != null`
2.  `getThingUID(RemoteDevice device)`  can return something different from null only if `getDetails!=null`, `getModelDetails!=null` and `getModelNumber!=null`. 
3. `getLabel(RemoteDevice device)`  is calling the getDisplayString() method which can never return null. It returns either empty or non-empty String. It returns an empty String if `getDetails==null` (but here we are already sure that it's not true) or if `getModelDetails==null && getManifacturerDetails==null)`  (but here we are already sure that it's not true, either)
4. So, basically the code in the [113-122] code line range can never be reached.  I tried to implement tests in order to cover it but I could not do it. getLabel(RemoteDevice device) can never return null, so in my opinion we can use just the `getDisplayString()` method.